### ISSUE TITLE
Results nnUNet train all fold with 18 subject

### ISSUE THREAD

**Issue by tzebre** [ivadomed/model-spinal-rootlets#7]
## 1) Dataset
I've predicted 8 subject from [ds004507/versions/1.0.1](https://openneuro.org/datasets/ds004507/versions/1.0.1) dataset with the model from [issue#5](https://github.com/ivadomed/model-spinal-rootlets/issues/5). I've review and corrected the prediction from these 8 images ground truth and the 12 already manually labeled. 

So now the dataset consists of 18 manually labeled images (or predicted with manual review). 
_I've dropped 2 subjects from the 10 images dataset, the quality was not sufficient._ 

While training the model, I started making ground truth of `sct-testing-large/sub-unfSCT* images. 

## 2) Model training/inference
nnUNet 3d_fullres model trained on 18 subject from [ds004507/versions/1.0.1](https://openneuro.org/datasets/ds004507/versions/1.0.1)
- Custom trained used: `nnUNetv2_train -tr nnUNetTrainer_250epochs`
- Prediction: `nnUNetv2_predict -d Dataset004_rootsegV2 -i IN_DIR -o OUT_DIR -f  0 1 2 3 4 -tr nnUNetTrainer_250epochs -c 3d_fullres -p nnUNetPlans`
- Postprocessing: ` nnUNetv2_apply_postprocessing ...`

[More documentation on nnUNet ](https://github.com/ivadomed/utilities/blob/main/quick_start_guides/nnU-Net_quick_start_guide.md)
### a) Score report:
- fold_0:
<img width="600" alt="Capture d’écran 2023-06-09 à 13 59 22" src="https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/d9814ebb-2bb4-4cd9-9b0c-42963a94413d">

- fold_1:
<img width="600" alt="Capture d’écran 2023-06-09 à 13 59 43" src="https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/af1eb628-4414-4465-ae43-d7ee7a97ec38">

- fold_2:
<img width="600" alt="Capture d’écran 2023-06-09 à 14 00 08" src="https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/4006a628-39f9-4208-aacc-b3e783180f98">

- fold_3:
<img width="600" alt="Capture d’écran 2023-06-09 à 14 00 26" src="https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/28879142-b0b2-4a4b-88d7-8e1eb8ac20ba">

- fold_4:
<img width="600" alt="Capture d’écran 2023-06-09 à 14 00 43" src="https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/6f63718d-584b-427b-b499-c5cd35cf0a5f">

### b) Images 

**RED = Manual ground truth**
**GREEN = Prediction**
**BLUE = Post processing**

#### Image from the [ds004507/versions/1.0.1](https://openneuro.org/datasets/ds004507/versions/1.0.1) dataset (used in training):
Dice score : 0.780776
<img width="700" alt="Capture d’écran 2023-06-09 à 14 16 55" src="https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/57424f8c-51c3-4002-abdb-86699d72dcb6">
<img width="700" alt="Capture d’écran 2023-06-09 à 14 17 22" src="https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/d2981249-7878-45fb-a589-ba979fd3c45d">
<img width="700" alt="Capture d’écran 2023-06-09 à 14 17 38" src="https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/11bdf618-ffbc-40a8-998a-fa09ccdd45fb">
<img width="700" alt="Capture d’écran 2023-06-09 à 14 18 03" src="https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/2d40ab1b-f84f-4808-9b90-46c0fc25ba4f">

#### Image from sct-testing-large/sub-unfSCT:
Dice score: 0.561508
<img width="700" alt="Capture d’écran 2023-06-09 à 14 21 05" src="https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/9a880421-41b1-436f-a332-a7aeed7d6dc9">
<img width="700" alt="Capture d’écran 2023-06-09 à 14 21 36" src="https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/a1762b19-055b-4397-b785-1bdcfad25777">
<img width="700" alt="Capture d’écran 2023-06-09 à 14 21 49" src="https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/6b2601e7-8fe4-4bcb-978a-ad4a3ed08061">
<img width="700" alt="Capture d’écran 2023-06-09 à 14 22 04" src="https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/17629c42-54fb-4db7-8315-b6548c72a46a">

## 3) Next steps 

- [x] Understand how full fold predictions work  
  - [folds and inference #1012 Q&A](https://github.com/MIC-DKFZ/nnUNet/discussions/1012#discussioncomment-2528578) 
    > Ensembling is implemented by averaging the softmax predictions 
- [ ] Look images in fold_4, and try to understand why the dice in lower than the other fold
- [ ] Try with nnUNet 3d_cascade 
- [x] Use this model to increase dataset size 
- [x] Try to improve ground truth with [denoising](https://github.com/ivadomed/model-spinal-rootlets/issues/5#issuecomment-1584852260)

**Comment by tzebre** [comment]
Prediction on 267 subject from [spine-generic/data-multi-subject](https://github.com/spine-generic/data-multi-subject) (r20230223) 
The images have been reoriented with sct_maths in LPI orientation. 

**sub-barcelona03**
![spine](https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/4dceadc1-e8cd-4e1f-8efd-38731412f693)

**sub-mountSinai01**
![spine](https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/4a13fcb9-16e0-41b4-8e9b-74b5eaa1ba65)
![spine](https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/ab8c0a62-2a7f-4b94-9967-62e70ea2f0ca)


Some predictions on lower cervical level seem to have wrong voxels. I'm manually reviewing some of the predictions to add subject to the training dataset.
